src/meili/map_matcher.cc

```diff
@@ -372,11 +372,12 @@ MapMatcher::OfflineMatch(const std::vector<Measurement>& measurements, uint32_t
   }
 
   //For k paths
-  std::vector<std::vector<MatchResult>> best_paths;
+  std::map<float, std::vector<MatchResult>> best_paths;
   for(uint32_t i = 0; i < k; ++i) {
     // Get the states for the kth best path its in reverse order
     std::vector<StateId> stateids;
     std::copy(vs_.SearchPath(time), vs_.PathEnd(), std::back_inserter(stateids));
+    auto accumulated_cost = vs_.AccumulatedCost(stateids.front());
```

Comment:
is this a reasonable way to get at how expensive it was to reach the end state?

Reply:
Yes, but the first stateid might be invalid; we need to find the first valid stateid.